src/mixin/TransientStorageArray.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.28;
+
+/// @title Transient Storage Array
+/// @author CoW DAO developers
+/// @notice Defines an interface to copy a (dynamic-size) bytes array from
+/// calldata to transient storage as well as to read it at a later point in the
+/// same transaction.
+/// @dev The array length is defined as a Solidity transient variable. The
+/// content of the array is stored word-by-word starting from the end of the
+/// transient storage space (slot `type(uint256).max`); each next word is stored
```

Comment:
Would it make sense to start writing into some weird random storage slot instead of right at the end to reduce the chance of interference?